src/python/platforms/fuchsia/device.py

```diff
@@ -16,6 +16,119 @@
 # TODO(mbarbella): Re-enable this check once functions below are implemented.
 # pylint: disable=unused-argument
 
+import os
+
+from fuchsia import errors
+from google_cloud_utils import gsutil
+from metrics import logs
+from system import environment
+from system import new_process
+from system import shell
+
+
+def qemu_setup():
+  """ Sets up and runs a QEMU VM in the background.
+  Does not block the calling process.
+  Fuchsia fuzzers assume a QEMU VM is running; call this routine prior to
+  beginning Fuchsia fuzzing tasks.
+  This initialization routine assumes that the GCS bucket contains the
+  standard Fuchsia SDK, as well as:
+  * /qemu-for-fuchsia/*
+  * /.ssh/* """
+  fuchsia_bucket_name = 'fuchsia_on_clusterfuzz_resources_v1'
+
+  resources_path = os.path.join(os.getcwd(), fuchsia_bucket_name)
+
+  shell.create_directory(resources_path)
+
+  fuchsia_resources_url = environment.get_value('FUCHSIA_RESOURCES_URL')
+  if fuchsia_resources_url is None:
+    raise errors.FuchsiaConfigError(
+        'Could not find path for remote'
+        'Fuchsia resources bucket (FUCHSIA_RESOURCES_URL')
+
+  gsutil_command_arguments = [
+      '-m', 'cp', '-r', fuchsia_resources_url, resources_path
+  ]
+  logs.log("Beginning Fuchsia SDK download.")
+  result = gsutil.GSUtilRunner().run_gsutil(gsutil_command_arguments)
+  if result.return_code or result.timed_out:
+    raise errors.FuchsiaSdkError('Failed to download Fuchsia'
+                                 'resources: ' + result.output)
+  logs.log("Fuchsia SDK download complete.")
+
+  # Save paths for necessary commands later.
+  qemu_path = os.path.join(resources_path, 'qemu-for-fuchsia', 'bin',
+                           'qemu-system-x86_64')
+  os.chmod(qemu_path, 0o550)
+  kernel_path = os.path.join(resources_path, 'target', 'x64', 'qemu-kernel.bin')
+  os.chmod(kernel_path, 0o644)
+  authorized_keys_path = os.path.join(resources_path, '.ssh', 'authorized_keys')
+  pkey_path = os.path.join(resources_path, '.ssh', 'pkey')
+  os.chmod(pkey_path, 0o400)
+  sharefiles_path = os.path.join(resources_path, 'qemu-for-fuchsia', 'share',
+                                 'qemu')
+
+  # The FVM is minimally sized to begin with; extend it to make room for
+  # ephemeral packages etc.
+  drive_path = os.path.join(resources_path, 'target', 'x64', 'fvm.blk')
+  os.chmod(drive_path, 0o644)
+  fvm_tool_path = os.path.join(resources_path, 'tools', 'fvm')
+  os.chmod(fvm_tool_path, 0o500)
+  process = new_process.ProcessRunner(fvm_tool_path,
+                                      [drive_path, 'extend', '--length', '1G'])
+  result = process.run_and_wait()
+  if result.return_code or result.timed_out:
+    raise errors.FuchsiaSdkError('Failed to extend FVM: ' + result.output)
+
+  # Need to bake keys into ZBI so we can SSH into it.  See:
+  # fuchsia.googlesource.com/fuchsia/+/refs/heads/master/sdk/docs/ssh.md
+  zbi_tool = os.path.join(resources_path, 'tools', 'zbi')
```

Comment:
could you break these different setup steps into separate functions? as is this function is getting a little too crowded and making things difficult to comprehend :)